### PR TITLE
[cleanup] test_tokenization_common.py

### PR DESCRIPTION
**Pull request by sshleifer**
- TesterMixin implements some oft repeated functions
- split long tests into multiple tests cases for better tracebacks
- misc typos
- type hints

### CODE REVIEW

File: tests/test_tokenization_common.py [hunk 1]
```diff
@@ -464,7 +466,9 @@ def test_special_tokens_mask(self):
         filtered_sequence = [x for x in filtered_sequence if x is not None]
         self.assertEqual(encoded_sequence, filtered_sequence)
 
-        # Testing with already existing special tokens
+    def test_special_tokens_mask_already_has_special_tokens(self):
```
Comment (sshleifer) [inline tests/test_tokenization_common.py:469]: split 1 big test into 3 small tests for better tracebacks

**Review by mfuntowicz — APPROVED** [review]
LGTM, thanks for taking care of this refactoring @sshleifer 💯

File: src/transformers/tokenization_utils.py [hunk 2]
```diff
@@ -2080,10 +2080,7 @@ def create_token_type_ids_from_sequences(self, token_ids_0: List, token_ids_1: O
     def build_inputs_with_special_tokens(self, token_ids_0: List, token_ids_1: Optional[List] = None) -> List:
         """
         Build model inputs from a sequence or a pair of sequence for sequence classification tasks
-        by concatenating and adding special tokens.
-        A RoBERTa sequence has the following format:
-            single sequence: <s> X </s>
-            pair of sequences: <s> A </s></s> B </s>
+        by concatenating and adding special tokens. This implementation does not add special tokens.
```
Comment (LysandreJik) [inline src/transformers/tokenization_utils.py:2083]: Nice catch

**Review by LysandreJik — APPROVED** [review]
Cool typings!

I like everything in this PR a lot.

**Pull request merged**